sdk/eventhub/azure-eventhub/azure/eventhub/_producer.py

```diff
@@ -185,41 +278,29 @@ def _on_outcome(self, outcome, condition):
 
     def _wrap_eventdata(
         self,
-        event_data,  # type: Union[EventData, EventDataBatch, Iterable[EventData]]
+        event_data_batch,  # type: EventDataBatch
         span,  # type: Optional[AbstractSpan]
         partition_key,  # type: Optional[AnyStr]
     ):
-        # type: (...) -> Union[EventData, EventDataBatch]
-        if isinstance(event_data, EventData):
-            if partition_key:
-                set_message_partition_key(event_data.message, partition_key)
-            wrapper_event_data = event_data
-            trace_message(wrapper_event_data, span)
-            add_link_to_send(wrapper_event_data, span)
-        else:
-            if isinstance(
-                event_data, EventDataBatch
-            ):  # The partition_key in the param will be omitted.
-                if (
-                    partition_key and partition_key != event_data._partition_key  # pylint: disable=protected-access
-                ):
-                    raise ValueError(
-                        "The partition_key does not match the one of the EventDataBatch"
-                    )
-                for message in event_data.message._body_gen:  # pylint: disable=protected-access
-                    add_link_to_send(message, span)
-                wrapper_event_data = event_data  # type:ignore
-            else:
-                if partition_key:
-                    event_data = _set_partition_key(event_data, partition_key)
-                event_data = _set_trace_message(event_data, span)
-                wrapper_event_data = EventDataBatch._from_batch(event_data, partition_key)  # type: ignore  # pylint: disable=protected-access
-        wrapper_event_data.message.on_send_complete = self._on_outcome
-        return wrapper_event_data
+        # type: (...) -> EventDataBatch
+        if (
+            partition_key and partition_key != event_data_batch._partition_key  # pylint: disable=protected-access
+        ):
+            raise ValueError(
+                "The partition_key does not match the one of the EventDataBatch"
+            )
+        for i in range(len(event_data_batch)):
+            event = event_data_batch.message._body_gen[i]  # pylint: disable=protected-access
+            add_link_to_send(event, span)
+            if self._enable_idempotent_partitions:
+                self._populate_idempotent_event_annotations(event, i)
+
+        event_data_batch.message.on_send_complete = self._on_outcome
+        return event_data_batch
 
     def send(
         self,
-        event_data,  # type: Union[EventData, EventDataBatch, Iterable[EventData]]
```

Comment:
This wasn't a public typehint right?

Reply:
nope, the `_producer` module is internal